code/modules/projectiles/guns/shotguns.dm

```diff
@@ -808,6 +808,66 @@ can cause issues with ammo types getting mixed up during the burst.
 /obj/item/weapon/gun/shotgun/pump/lever/pump_notice(mob/user)
 	to_chat(user, span_notice("<b>You work [src] lever.</b>"))
 
+/obj/item/weapon/gun/shotgun/pump/lever/unload(mob/user)
+	if(pump_lock)
+		to_chat(user, span_notice("<b>You pull [src]'s lever downward, ejecting the cartridge.</b>"))
+		pump_lock = FALSE //we're operating the slide release to unload, thus unlocking the pump
+	return ..()
+// ***********************************************
+// Winchester Rifle. The gun that won the west.
+
+/obj/item/weapon/gun/shotgun/pump/repeater
```

Comment:
you've put this in the middle of the lever proc, put that in the end of the file for exemple